src/test/java/ru/innopolis/stc12/booksharing/model/dao/BookEditionsDaoImplTest.java

```diff
@@ -55,22 +56,37 @@ void getAllBookEditions() {
 
     @Test
     void getBookEditionByIsbn() {
-        BookEdition bookEdition = new BookEdition("title", "desc", "isbn");
-        when(jdbcTemplate.queryForObject(anyString(), any(Object[].class),
-                any(RowMapper.class))).thenReturn(bookEdition);
-        assertEquals(bookEdition, bookEditionsDao.getBookEditionByIsbn("22325235"));
+        List<BookEdition> list = new ArrayList<>();
+        list.add(new BookEdition());
+        when(jdbcTemplate.query(anyString(), any(Object[].class),
+                any(RowMapper.class))).thenReturn(list);
+        assertEquals(list.get(0), bookEditionsDao.getBookEditionByIsbn("123"));
```

Comment:
зачем нам тут лист создавать? isbn уникальный номер, и ты получишь в итоге или один экземпляр издания или ничего

Reply:
Если я верно понял, то queryForObject должен обязательно вернуть 1 строку, если ничего не находит, то вызывает exception. Почитал пару похожих запросов на форумах, рекомендовали пользоваться  query, но он возвращает только List.

Reply:
@rmunirov вроде

> Returns:
> the result object of the required type, or null in case of SQL NULL

[queryForObject ](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/jdbc/core/JdbcTemplate.html#queryForObject-java.lang.String-java.lang.Class-)